src/gwt/src/org/rstudio/studio/client/workbench/UserInterfaceHighlighter.java

```diff
@@ -70,19 +91,109 @@ public Element getHighlightElement()
       {
          return highlightedElement_;
       }
-      
+
+      public void clearHandler()
+      {
+         if (handler_ != null)
+            handler_.removeHandler();
+      }
+
+      public void executeCallback()
+      {
+         // This method must be called by a single HighlightPair, but because there can be multiple
+         // pairs per query, we need to check if the callback has already executed before proceeding.
+         if(!highlighter_.getCallbackProcessed(index_))
+         {
+            highlighter_.setCallbackProcessed(index_, true);
+            highlighter_.getServer().executeRCode(callback_, new ServerRequestCallback<String>(){
+
+               @Override
+               public void onResponseReceived(String results)
+               {
+                  // Remove listener from this element and all other elements with the same query
+                  highlighter_.clearEvents();
+               }
+
+               @Override
+               public void onError(ServerError error)
+               {
+                  Debug.logError(error);
+               }
+            });
+         }
+      }
+
+      private HandlerRegistration addListener()
+      {
+         final JavaScriptObject functor = addEventListener(callback_, monitoredElement_);
```

Comment:
nit: This is really just a plain JavaScript function object rather than something like a functor (e.g. a callable instantiation of some class)

Reply:
Fixed :)